marshmallow/fields.py

```diff
@@ -1083,10 +1086,12 @@ class TimeDelta(Field):
 
     def __init__(self, precision='seconds', error=None, **kwargs):
         precision = precision.lower()
-        units = (self.DAYS, self.SECONDS, self.MICROSECONDS)
+        units = (self.DAYS, self.SECONDS, self.MICROSECONDS, self.MILLISECONDS,
+                 self.MINUTES, self.HOURS, self.WEEKS)
 
         if precision not in units:
-            msg = 'The precision must be "{0}", "{1}" or "{2}".'.format(*units)
+            msg = 'The precision must be "{0}" or "{1}".'.format(
```

Comment:
The quotation marks don't display quite right here:

```
The precision must be "days, seconds, microseconds, milliseconds, minutes, hours" or "weeks".'
```

I can fix this when I merge.